### ISSUE TITLE
Terraform Provider for Organization Resources

### ISSUE THREAD

**Issue by jmprovencher** [PrefectHQ/prefect#7679]
### First check

- [X] I added a descriptive title to this issue.
- [X] I used the GitHub search to find a similar request and didn't find it.
- [X] I searched the Prefect documentation for this feature.

### Prefect Version

2.x

### Describe the current behavior

Currently, the only way to create resources such as workspaces and api keys are through the UI or through the api calls.

### Describe the proposed behavior

It would be very interesting to have a terraform provider that allows customer to manage their resource lifecycle through terraform. If any manual operations happened, or a workspace is deleted, it would automatically recreated for example. It's also a pretty interesting way of managing service accounts for Enterprise customers.

### Example Use

resource "prefect_workspace" "my_workspace" {
     name = "development"
}

**Comment by peytonrunyan** [comment]
@gabcoyne Do you have any thoughts on this or know who the right person to ask about this? I know next to nothing about IAC.

**Comment by gabcoyne** [comment]
Yep we have a WIP, check out [this repo](https://github.com/PrefectHQ/terraform-provider-prefect)

**Comment by panasenco** [comment]
+1 for this, we definitely desperately need this to manage infrastructure blocks, storage blocks, service accounts, and work queues.

**Comment by cicdw** [comment]
Closing as https://github.com/PrefectHQ/terraform-provider-prefect is now in production for many folks - for fastest response times on enhancement requests / bug reports, please use that repo.
